doc/mgmt/SONiC_Ztp_DesignDoc.md

```diff
@@ -0,0 +1,254 @@
+# Zero Touch Provisioning (ZTP)
+ZTP support through sonic-management framework
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev  |    Date    |      Author       | Change Description |
+| :--: | :--------: | :---------------: | ------------------ |
+| 0.1  | 10/10/2019 | Arunsundar Kannan | Initial version    |
+
+# About this Manual
+This document provides general information about the ZTP support inside SONiC management framework.
+# Scope
+Covers Northbound interface for the ZTP feature, as well as Unit Test cases.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term** | **Meaning**             |
+| -------- | ----------------------- |
+| ZTP      | Zero Touch Provisioning |
+
+# 1 Feature Overview
+
+Provide management framework capabilities to handle:
+
+- Show status of ztp service
+- Enable ztp service
+- Disable ztp service
+- Run ztp service
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide management framework support to existing SONiC capabilities with respect to ZTP
+
+### 1.1.2 Configuration and Management Requirements
+- CLI style configuration and show commands
+- REST API support
+- gNMI Support
+
+### 1.1.3 Scalability Requirements
+N/A
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+The feature will use the traditional translib/transformer approach with an addition of a service running in the host which does all the "actual work". The service will use the existing sonic ztp scripts in the host.
+
+### 1.2.2 Container
+Management Container
+
+### 1.2.3 SAI Overview
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+**show ztp-status**
+
+This command displays the current ZTP configuration of the switch. It also displays detailed information about current state of a ZTP session. It displays information related to all configuration sections as defined in the switch provisioning information discovered in a particular ZTP session.
+
+**ztp enable**
+
+The *ztp enable* command is used to administratively enable ZTP. When ZTP feature is included as a build option, ZTP service is configured to be enabled by default. This command is used to re-enable ZTP after it has been disabled by user. It is to be noted that this command will only modify the ZTP configuration file and does not perform any other actions.
```

Comment:
Instead of italicizing "ztp enable", enclose it in backticks so that it looks like `ztp enable`